tests/helpers/cache_syncing_client.go

```diff
@@ -19,6 +20,13 @@ func NewCacheSyncingClient(client client.Client) *CacheSyncingClient {
 	}
 }
 
+func NewCacheSyncingClientFromConfig(config *rest.Config, options client.Options) *CacheSyncingClient {
+	k8sClient, err := client.New(config, options)
```

Comment:
Are you sure we are going to get a client with cache configured here? We only pass in scheme in the options. I _think_ we need some cache configuration too.

Reply:
Well, as this is a test utility, I don't think we should be really concerned about it. The cache syncing client does not deal with a cache directly, instead it only awaits that the result of an operation is visible by the client (maybe naming is probably not great, suggestion on naming are welcome). It does not really matter whether a cache is actually used or not.

Reply:
Oh, I think I see what you mean - this admin client that I mentioned above would need to share controllers cache, so that stuff that has been done with it is immediately visible by the controllers. This would not be the case here as this client shares no cache. Did I get your point?

Reply:
I have addressed the above by changing the method to receive the manager cache so that it can be shared.